lib/intercept.js

```diff
@@ -388,7 +388,17 @@ function activate() {
       options = urlParse(options.toString())
     }
     options.proto = proto
-
```

Comment:
As I mentioned I'm not sure we need to insert `getHeader` into `options` at all. I've found no evidence that this is for parity with an existing API or compatibility with some existing library.

Could you remove this code, and update the code in the places where `options.getHeader` is called, instead? It's fine to delegate the repeated code to a helper function, though let's not unnecessarily stick that helper function into `options`.